src/app/app.ts

```diff
@@ -36,12 +39,70 @@ export class App {
     this.characterService.characters(),
   );
 
+  protected dropTargetIndex = signal<number | null>(null);
+  protected draggedItemIndex = signal<number | null>(null);
+  protected isDragging = signal<boolean>(false);
+
   protected readonly title = 'My Personal Top Characters';
   protected readonly characters = this._characters.asReadonly();
 
-  protected onDrop(event: CdkDragDrop<Character[]>): void {
-    const characters = this._characters();
-    moveItemInArray(characters, event.previousIndex, event.currentIndex);
-    this._characters.set([...characters]);
+  protected onDragStarted(index: number): void {
+    this.draggedItemIndex.set(index);
+    this.isDragging.set(true);
+  }
+
+  protected onDragEnded(): void {
+    this.dropTargetIndex.set(null);
+    this.draggedItemIndex.set(null);
+    this.isDragging.set(false);
+  }
+
+  protected onDragMoved(event: CdkDragMove): void {
+    const previewY = event.pointerPosition.y;
+
+    const separators = document.querySelectorAll('.drop-separator.visible');
+    let targetIndex: number | null = null;
+    let minDistance = Infinity;
+
+    separators.forEach((separator, index) => {
```

Comment:
another option is to use IntersectionObservers for detection, but that solution is also fine

Reply:
Thanks for the tip, haven't come across him yet